Packaging-Roadmap-for-Mantl.md

```diff
@@ -0,0 +1,145 @@
+# Packaging Roadmap for Mantl
+#### Rationale
+To improve mantl's deployment speed, we are going to replace some
+ansible role logic with packages. These packages will be defined by a hammer spec,
+and uploaded to bintray. Anything that is as simple as 'copy this file here'
+or 'run this command' can be ported to these packages. Then, the ansible roles can
+be updated to simple package install commands
+
+Because we are using the hammer tool to build these, when hammer supports `.deb` packages,
+ubuntu/debian support will come much more easily.
+
+Logic that relies upon Jinja2 templates could be ported to Consul templates,
+and those could be ported to packages.
+
+The more ansible roles that we port to packages, the faster the build process will be.
+If we can remove ansible entirely, we can have mantl bootstrap from terraform, and
+make cluster deployment into a one-step process.
+
+#### Ansible roles -> New Packages
+Here is the list from the `terraform.sample.yml` ansible playbook. For each role that
+we can port to a package, put the package name after it, and list the package description
+below, or make a note. I hope that the ad-hoc formatting here makes sense.
+
+I'm not going to include any package upgrade logic here, either for yum or pip.
+We will let the user/packages decide what versions to use
+
+Another assumption here is that each of these packages are going to have
+distributive check files associated with them. As of right now, those are in
+the distributive-{component} packages.
+
+*Roles for all hosts*
+- common: **mantl-common**
+  - System Dependencies
+    - python-pip
+    - httpd-tools
+    - nc
+    - openssh
+    - policycoreutils-python
+    - epel-release
+    - unzip
+  - Ansible defaults: selinux with permissive policy and central configuration
+  - Ansible handlers: update-ca-trust -> run `update-ca-trust` cmd in package script when needed
+  - Ansible main tasks
+    - set timezone to UTC -> `ln -sf /etc/localtime /usr/share/zoneinfo/Etc/UTC`
+    - create `/etc/mantl` to hold metadata for state of cluster pre-consul boot
+    - j2 template for `/etc/hosts` -> consul template for `/etc/hosts`
```

Comment:
Maybe we can skip the host population and add .node.consul to the dns search path.